tests/expressions_case/tests.py

```diff
@@ -0,0 +1,1136 @@
+                (2, UUID('22222222222222222222222222222222')), (3, None), (3, None), (4, None)
+            ],
+            transform=attrgetter('integer', 'uuid')
+        )
+
+    def test_update_fk(self):
+        obj1, obj2 = CaseTestModel.objects.all()[:2]
+
+        CaseTestModel.objects.update(
+            fk=Case(
+                When(integer=1, then=Value(obj1.pk)),
+                When(integer=2, then=Value(obj2.pk))
+            )
+        )
+
+        self.assertQuerysetEqual(
+            CaseTestModel.objects.all().order_by('pk'),
+            [(1, obj1.pk), (2, obj2.pk), (3, None), (2, obj2.pk), (3, None), (3, None), (4, None)],
+            transform=attrgetter('integer', 'fk_id')
+        )
+
+    def test_lookup_in_condition(self):
+        self.assertQuerysetEqual(
+            CaseTestModel.objects.annotate(
+                test=Case(
+                    When(integer__lt=2, then=Value('less than 2')),
+                    When(integer__gt=2, then=Value('greater than 2')),
+                    default=Value('equal to 2'),
+                    output_field=models.CharField()
+                )
+            ).order_by('pk'),
+            [
+                (1, 'less than 2'), (2, 'equal to 2'), (3, 'greater than 2'), (2, 'equal to 2'), (3, 'greater than 2'),
+                (3, 'greater than 2'), (4, 'greater than 2')
+            ],
+            transform=attrgetter('integer', 'test')
+        )
+
+    def test_lookup_different_fields(self):
+        self.assertQuerysetEqual(
+            CaseTestModel.objects.annotate(
+                test=Case(
+                    When(integer=2, integer2=3, then=Value('when')),
+                    default=Value('default'),
+                    output_field=models.CharField()
+                )
+            ).order_by('pk'),
+            [
+                (1, 1, 'default'), (2, 3, 'when'), (3, 4, 'default'), (2, 2, 'default'), (3, 4, 'default'),
+                (3, 3, 'default'), (4, 5, 'default')
+            ],
+            transform=attrgetter('integer', 'integer2', 'test')
+        )
+
+    def test_combined_q_object(self):
+        self.assertQuerysetEqual(
+            CaseTestModel.objects.annotate(
+                test=Case(
+                    When(Q(integer=2) | Q(integer2=3), then=Value('when')),
+                    default=Value('default'),
+                    output_field=models.CharField()
+                )
+            ).order_by('pk'),
+            [
+                (1, 1, 'default'), (2, 3, 'when'), (3, 4, 'default'), (2, 2, 'when'), (3, 4, 'default'),
+                (3, 3, 'when'), (4, 5, 'default')
+            ],
+            transform=attrgetter('integer', 'integer2', 'test')
+        )
+
+
+class CaseDocumentationExamples(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        Client.objects.create(
+            name='Jane Doe',
+            account_type=Client.REGULAR,
+            registered_on=date.today() - timedelta(days=36)
```

Comment:
If you have time, we also try to include on trailing comma on the last kwarg so if more items are added later, we don't have to modify that line again.